diff -urN dosemu-1.4.0.7/src/base/init/lexer.h dosemu-1.4.0.7-patched/src/base/init/lexer.h
--- dosemu-1.4.0.7/src/base/init/lexer.h	2012-07-08 07:38:39.000000000 +1100
+++ dosemu-1.4.0.7-patched/src/base/init/lexer.h	2012-11-23 03:17:16.576334869 +1100
@@ -36,5 +36,5 @@
 extern int yylex_destroy(void);
 extern FILE* yyget_in(void);
 extern FILE* yyget_out(void);
-extern int yyget_leng(void);
+extern size_t yyget_leng(void);
 extern char *yyget_text(void);
