--- src/base/init/init.c    (Revision 1862)
+++ src/base/init/init.c    (Revision 1863)
@@ -262,10 +262,18 @@
 
   if (result != NULL)
     {
+      int err = errno;
       perror ("LOWRAM mmap");
-      config.exitearly = 1;
+      if (err == EINVAL) {
+    fprintf(stderr, "Cannot map low DOS memory (the first 640k).\n"
+        "You can most likely avoid this problem by running\n"
+        "sysctl -w vm.mmap_min_addr=0\n"
+        "as root, or by changing the vm.mmap_min_addr setting in\n"
+        "/etc/sysctl.conf to 0.\n");
     }
+      leavedos(99);
 }
+}
 
 /*
  * DANG_BEGIN_FUNCTION version_init
